core-common/src/main/java/org/mapstruct/Context.java

```diff
@@ -25,14 +25,22 @@
 
 /**
  * Marks a parameter of a method to be treated as <em>mapping context</em>. Such parameters are passed to other mapping
- * methods, {@link ObjectFactory} methods or {@link BeforeMapping}/{@link AfterMapping} methods when applicable and can
- * thus be used in custom code. The {@link Context} parameters are otherwise ignored by MapStruct.
+ * methods, {@code @}{@link ObjectFactory} methods or {@code @}{@link BeforeMapping}/{@code @}{@link AfterMapping}
+ * methods when applicable and can thus be used in custom code.
  * <p>
- * For generated code to call a method that is declared with {@link Context} parameters, the declaration of the mapping
- * method being generated needs to contain at least those (or assignable) {@link Context} parameters as well. MapStruct
- * will not create new instances of missing {@link Context} parameters nor will it pass {@code null} instead.
+ * The type of an {@code @Context} parameter is also inspected for
+ * {@code @}{@link BeforeMapping}/{@code @}{@link AfterMapping} methods, which are called on the provided context
+ * parameter value if applicable.
  * <p>
- * Example:
+ * <strong>Note:</strong> no {@code null} checks are performed before calling before/after mapping methods on context
+ * parameters. The caller needs to make sure that no {@code null} are passed in that case.
+ * <p>
+ * For generated code to call a method that is declared with {@code @Context} parameters, the declaration of the mapping
+ * method being generated needs to contain at least those (or assignable) {@code @Context} parameters as well. MapStruct
+ * will not create new instances of missing {@code @Context} parameters nor will it pass {@code null} instead.
+ * <p>
+ * <strong>Example 1:</strong> Using {@code @Context} parameters for passing data down to hand-written property mapping
+ * methods and {@code @} {@link BeforeMapping} methods:
```

Comment:
Minor detail: the "@" seems to be handled a bit inconsistently. So far we always expressed it via "&#64;". Here I probably wouldn't bother adding it at all.